pkg/apiserver/resthandler.go

```diff
@@ -243,6 +244,96 @@ func UpdateResource(r RESTUpdater, ctxFn ContextFunc, namer ScopeNamer, codec ru
 	}
 }
 
+// PatchResource returns a function that will handle a resource patch
```

Comment:
This borrows heavily from GetResource and UpdateResource. Some shared routines could be broken out but this would break the conventional style of each handler being an individual "main". I'm not sure if this style was intended.